container-storage-setup.sh

```diff
@@ -437,6 +437,46 @@ check_docker_storage_metadata() {
   exit 1
 }
 
+# See whether we can use overlay backend
+can_do_overlay() {
+    if test -d $_DOCKER_METADATA_DIR; then
+        mountroot=$_DOCKER_METADATA_DIR
+    else
+        mountroot=/var
+    fi
+    fstype=$(stat -f -c '%T' $mountroot)
+    have_ftype=yes
+    case "$fstype" in
+        # For xfs, see https://bugzilla.redhat.com/show_bug.cgi?id=1288162#c8
+        xfs) if test "$(xfs_info $mountroot | grep -o 'ftype=[01]')" = "ftype=0"; then
+                 have_ftype=no
+                 Info "XFS filesystem at ${mountroot} has ftype=0, cannot use overlay backend; consider separate volume (or OS reprovision)"
+             fi
+             ;;
+       ext4) # I'm just going to assume this is OK, but we should probably try to check it?
```

Comment:
I think ext4 and others case can be merged into one. No need to special case ext4, as of now.